managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -2156,47 +2156,55 @@ void readOperationCompleted() {
     }
 
     void asyncDeleteLedger(final LedgerHandle lh) {
-        if (lh == null) {
+        asyncDeleteLedger(lh, 3);
```

Comment:
Can you take out the `3` into a constant `DEFAULT_RETRIES_LEDGER_DELETE`?

Reply:
Also, should the retry be happening also for data ledgers?

Reply:
> Also, should the retry be happening also for data ledgers?

yes, I will fix it.